-1

Tested at: 580542555863e8512e5f7a15bcbb5745da3a86b3

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25294/31744/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found compilation error when building: 

<pre>>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_4_X_2018-11-19-1100/src/SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizer.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_4_X_2018-11-19-1100/src/SimTracker/SiPixelDigitizer/plugins/PreMixingSiPixelWorker.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_4_X_2018-11-19-1100/src/SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizerAlgorithm.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_4_X_2018-11-19-1100/src/SimTracker/SiPixelDigitizer/plugins/SealModule.cc 
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_4_X_2018-11-19-1100/src/SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizer.cc: In member function 'virtual void cms::SiPixelDigitizer::finalizeEvent(edm::Event&, const edm::EventSetup&)':
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_4_X_2018-11-19-1100/src/SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizer.cc:254:38: error: no match for 'operator=' (operand types are 'std::unique_ptr<PileupMixingContent>' and 'PileupMixingContent*')
     PileupInfo_ = getEventPileupInfo();
                                      ^
In file included from /cvmfs/cms-ib.cern.ch/nweek-02551/slc6_amd64_gcc700/external/gcc/7.0.0-omkpbe2/include/c++/7.3.1/memory:80:0,
                 from /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_4_X_2018-11-19-1100/src/SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizer.cc:22:
/cvmfs/cms-ib.cern.ch/nweek-02551/slc6_amd64_gcc700/external/gcc/7.0.0-omkpbe2/include/c++/7.3.1/bits/unique_ptr.h:281:7: note: candidate: std::unique_ptr<_Tp, _Dp>& std::unique_ptr<_Tp, _Dp>::operator=(std::unique_ptr<_Tp, _Dp>&&) [with _Tp = PileupMixingContent; _Dp = std::default_delete<PileupMixingContent>]
</pre>

* **Clang**:

I found compilation warning while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 16 COMPILER='llvm compile'

```
 See details on the summary page.

--- a/SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizer.cc
+++ b/SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizer.cc
@@ -251,13 +251,12 @@ namespace cms
     std::vector<edm::DetSet<PixelDigi> > theDigiVector;
     std::vector<edm::DetSet<PixelDigiSimLink> > theDigiLinkVector;
  
-    PileupInfo_ = getEventPileupInfo();
     if (firstFinalizeEvent_) {
       const unsigned int bunchspace = PileupInfo_->getMix_bunchSpacing();
       _pixeldigialgo->init_DynIneffDB(iSetup, bunchspace);
       firstFinalizeEvent_ = false;
     }
-    _pixeldigialgo->calculateInstlumiFactor(PileupInfo_);   
+    _pixeldigialgo->calculateInstlumiFactor(PileupInfo_.get());
 
     for( const auto& iu : pDD->detUnits()) {
       

--- a/SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizer.h
+++ b/SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizer.h
@@ -62,10 +62,10 @@ namespace cms {
 					 std::vector<int> &bunchCrossingList,
 					 std::vector<float> &TrueInteractionList, 
 					 std::vector<edm::EventID> &eventInfoList, int bunchSpacing) override{
-      PileupInfo_ = new PileupMixingContent(numInteractionList, bunchCrossingList, TrueInteractionList, eventInfoList, bunchSpacing);
+      PileupInfo_ = std::make_unique<PileupMixingContent>(numInteractionList, bunchCrossingList, TrueInteractionList, eventInfoList, bunchSpacing);
     }
 
-    PileupMixingContent* getEventPileupInfo() override { return PileupInfo_; }
+    PileupMixingContent* getEventPileupInfo() override { return PileupInfo_.get(); }
 
   private:
     void accumulatePixelHits(edm::Handle<std::vector<PSimHit> >,
@@ -95,7 +95,7 @@ namespace cms {
     std::map<unsigned int, PixelGeomDetUnit const *> detectorUnits;
     CLHEP::HepRandomEngine* randomEngine_ = nullptr;
 
-    PileupMixingContent* PileupInfo_;
+    std::unique_ptr<PileupMixingContent> PileupInfo_;
     
     const bool pilotBlades; // Default = false
     const int NumberOfEndcapDisks; // Default = 2

--- a/SimTracker/SiStripDigitizer/plugins/SiStripDigitizer.cc
+++ b/SimTracker/SiStripDigitizer/plugins/SiStripDigitizer.cc
@@ -154,8 +154,7 @@ void SiStripDigitizer::accumulateStripHits(edm::Handle<std::vector<PSimHit> > hS
     const TrackerTopology *tTopo=tTopoHand.product();
 
     //Re-compute luminosity for accumulation for HIP effects
-    PileupInfo_ = getEventPileupInfo();
-    theDigiAlgo->calculateInstlumiScale(PileupInfo_);
+    theDigiAlgo->calculateInstlumiScale(PileupInfo_.get());
 
     // Step A: Get Inputs
     for(auto const& trackerContainer : trackerContainers) {

--- a/SimTracker/SiStripDigitizer/plugins/SiStripDigitizer.h
+++ b/SimTracker/SiStripDigitizer/plugins/SiStripDigitizer.h
@@ -56,10 +56,10 @@ public:
 				       std::vector<int> &bunchCrossingList,
 				       std::vector<float> &TrueInteractionList,
 				       std::vector<edm::EventID> &eventInfoList, int bunchSpacing) override{
-    PileupInfo_ = new PileupMixingContent(numInteractionList, bunchCrossingList, TrueInteractionList, eventInfoList, bunchSpacing);
+    PileupInfo_ = std::make_unique<PileupMixingContent>(numInteractionList, bunchCrossingList, TrueInteractionList, eventInfoList, bunchSpacing);
   } 
 
-  PileupMixingContent* getEventPileupInfo() override { return PileupInfo_; } 
+  PileupMixingContent* getEventPileupInfo() override { return PileupInfo_.get(); }
 
   
 private:
@@ -100,7 +100,7 @@ private:
   CLHEP::HepRandomEngine* randomEngine_ = nullptr;
   std::vector<std::pair<int,std::bitset<6>>> theAffectedAPVvector;
 
-  PileupMixingContent* PileupInfo_;
+  std::unique_ptr<PileupMixingContent> PileupInfo_;
 
 };
 

